models/llama/llama/math_ops.py

```diff
@@ -70,14 +71,15 @@ def attention(self, xq, keys, values, head_dim, mask):
 
     @Profiler.profiling_decorator("softmax")
     def softmax(self, x, dim):
-        if self.use_triton:
-            return F.softmax(x, dim=-1)
+        if self.use_triton and len(x) == 2:
+            return triton_softmax(x, dim=-1)
```

Comment:
Why are we passing `dim=-1` to these calls, when we receive `dim` as an argument?  Let's pass it through properly instead of overriding it.  (Also, does the fused Triton kernel actually handle `dim!=-1` correctly?)

Reply:
Currently it does not handle  dim != -1 . Looking into it (seeing how llama.cpp is doing this) if you have any pointers.